### DISCUSSION TITLE
GUI of Renode is not launching

### DISCUSSION THREAD

**Discussion by shannon-Weber** [polarfire-soc/polarfire-soc-documentation#147] [Applications and demos] [answered]
[WARNING] Couldn't start UI - falling back to telnet mode
this is related to #141

**Comment by AntonKrug** [comment, marked as answer]
Sorry I never had this issue, so I do not have direct answer with a solution. 
But this is just my speculation, maybe you have some package missing which is needed for the GUI.
It might be installed as its own 32-bit variant, so for some applications it might ok and present, while for some (renode/mono) it might not be present. I notice the warning before in the previous issue that you might have installed some 32-bit libs.
For that probably you added the 32-bit architecture to yours at some point in the past (with a command like this `dpkg --add-architecture i386`, do not run it, i'm just listing it here to see if it will ring any bell)

My suspition is that when you run the `dpkg --print-architecture` command, it will print not just amd64 but i386 too.

Then if you could run this `dpkg -l | awk '/^ii/ && $4 == "i386" { print }'` it should print all the 32-bit installed packages. 

If the output will be empty or listing just few packages then probably I'm wrong, but if the output will be very long with a lots of packages then that could be a indication or maybe even a cause of all sorts of issues. I have many years ago a one Debian installation which was using heavily 32-bit packages on a 64-bit OS and it was cause of all sorts of troubles.

**Reply by AntonKrug** [reply]
Another thing you could try is to run Renode directly (not from SoftConsole) just to rule out any SoftConsole influence.

**Reply by shannon-Weber** [reply]
https://github.com/renode/renode/blob/master/README.rst#installation 

From the above link installed renode and dependency , GUI is coming now.

**Reply by AntonKrug** [reply]
Yes the `gtk-sharp2` and `libgtk2.0-0` are needed dependencies, but usually interactive GUI desktop ubuntu installations have that already. So maybe your installation had it as a i386 package till now and the amd64 version was missing for renode till now.

**Comment by ThibFrgsGmz** [comment]
I don't know if this helps, but I already encountered this message when I was running Renode manually on WSL-2.

**Reply by AntonKrug** [reply]
On WSL you have very stripped distributions compared to interactive GUI desktop installs which are bloated with packages. For that case I would see WSL almost on-par with docker containers. For example here is a debian based container and installing packages needed for Renode:

https://github.com/AntonKrug/docker-images/blob/master/softconsole-base-slim/Dockerfile#L29-L43

**Reply by AntonKrug** [reply]
The GH multi-line highlighting sometimes work, sometimes doesn't. I was highlighting lines 29 to 43.

**Comment by shannon-Weber** [comment]
Hi AntonKrug , not sure which 32 bit one is creating problem  
![image](https://user-images.githubusercontent.com/109544441/180822930-e852dc35-5604-4217-b631-ea810bf80243.png)
